src/builder/build_traits.rs

```diff
@@ -29,17 +30,40 @@ use crate::Hugr;
 
 use crate::hugr::HugrMut;
 
+pub trait Buildable {
+    type BaseMut<'a>: Buildable + HugrMut
+    where
+        Self: 'a;
+    type BaseView<'a>: HugrView
+    where
+        Self: 'a;
+    /// The underlying [`Hugr`] being built
+    fn hugr_mut(&mut self) -> Self::BaseMut<'_>;
+    /// Immutable reference to HUGR being built
+    fn hugr(&self) -> Self::BaseView<'_>;
+}
+
+impl<T: AsMut<Hugr> + AsRef<Hugr>> Buildable for T {
```

Comment:
An interesting alternative (in that it illustrates what's going on) would be one `impl Buildable for Hugr` and a second `impl<T: HugrMut + Copy> Buildable for T` - as every `&mut` is Copy (right?!). That's two `impl` blocks rather than one though so I didn't...